app/src/main/kotlin/com/blacksquircle/ui/internal/di/DataModule.kt

```diff
@@ -51,8 +53,12 @@ object DataModule {
     @Provides
     @Singleton
     @Named("Local")
-    fun provideLocalFilesystem(): Filesystem {
-        return FilesystemDelegate.provideFilesystem(Environment.getExternalStorageDirectory())
+    fun provideLocalFilesystem(sharedPreferences: SharedPreferences): Filesystem {
+        var localSaveddir: String = sharedPreferences.getString(KEY_LOCAL_SAVEDDIR, "") ?: ""
+        if (localSaveddir.equals(""))
+            return FilesystemDelegate.provideFilesystem(Environment.getExternalStorageDirectory())
+        else
+            return FilesystemDelegate.provideFilesystem(File(localSaveddir))
```

Comment:
I don't remember exactly, do you able to open the previous tab (root folder) after relaunch?